### PR TITLE
Fixes #8710 - introduced foreman-debug subpackage

### PR DESCRIPTION
**Pull request by lzap**
Is this kosher?

### CODE REVIEW

**Comment by domcleal** [comment]
Yeah, that's OK.  Can you do the same for the debs?

**Comment by domcleal** [comment]
+ comps update

**Comment by lzap** [comment]
Sure, I was thinking to require this package by foreman-proxy. What do you think?

Also as my next step, I want to add foreman-proxy support (files which are only on proxy + underlaying services).

**Comment by domcleal** [comment]
I guess that's OK - the dependency seems slightly strange, but I can't think of a reason why it'd be a problem.

**Comment by lzap** [comment]
Pushed the comps as well as proxy part.

Foreman: http://koji.katello.org/koji/taskinfo?taskID=218303
Proxy: http://koji.katello.org/koji/taskinfo?taskID=218309

**Comment by eLobato** [comment]
foreman & smart-proxy pieces are now merged

File: foreman-proxy/foreman-proxy.spec [hunk 1]
```diff
@@ -184,6 +188,7 @@ rm -rf %{buildroot}
 %{_unitdir}/%{name}.service
 %{_prefix}/lib/tmpfiles.d/%{name}.conf
 %endif
+/usr/share/foreman/script/foreman-debug.d/75-foreman-proxy
```
Comment (domcleal) [inline foreman-proxy/foreman-proxy.spec]: _datadir, ditto above

File: foreman/foreman.spec [hunk 2]
```diff
@@ -196,6 +199,17 @@ Meta Package to install hammer rubygems and its dependencies
 
 %files cli
 
+%package debug
+Summary: Foreman Debug utilities
+Group: Applications/System
+
+%description debug
+Useful utilities for debug info collecting
```
Comment (domcleal) [inline foreman/foreman.spec]: collection

File: foreman/foreman.spec [hunk 3]
```diff
@@ -196,6 +199,17 @@ Meta Package to install hammer rubygems and its dependencies
 
 %files cli
 
+%package debug
+Summary: Foreman Debug utilities
```
Comment (domcleal) [inline foreman/foreman.spec]: neither or both of "debug" and "utilities" should be capitalised

File: foreman/foreman.spec [hunk 4]
```diff
@@ -563,6 +577,8 @@ rm -rf %{buildroot}
 %dir %{_datadir}/%{name}
 %{_datadir}/%{name}/app
 %exclude %{_datadir}/%{name}/app/assets
+%exclude %{_datadir}/%{name}/bundler.d/*
+%exclude %{_datadir}/%{name}/script/%{name}-debug.d
```
Comment (domcleal) [inline foreman/foreman.spec]: can you rebase this since I changed how %files works?

**Comment by lzap** [comment]
All fixed sorry for the delay. http://koji.katello.org/koji/taskinfo?taskID=220938

**Comment by lzap** [comment]
Is there anything else than this PR that blocks the branching/release?

**Comment by domcleal** [comment]
@lzap this PR and we need an equivalent for deb/develop

File: foreman/foreman.spec [hunk 5]
```diff
@@ -559,6 +573,8 @@ rm -rf %{buildroot}
 %dir %{_datadir}/%{name}
 %{_datadir}/%{name}/app
 %exclude %{_datadir}/%{name}/app/assets
+%exclude %{_datadir}/%{name}/bundler.d/*
```
Comment (domcleal) [inline foreman/foreman.spec]: remove this line, not required

File: foreman/foreman.spec (hunk 4 above)
Comment (domcleal) [inline foreman/foreman.spec]: This dir isn't in the package, is it?

Comment (domcleal) [inline foreman/foreman.spec]: Ack, never mind, I see it now.

**Comment by domcleal** [comment]
Do you have a scratch build of foreman-proxy too please?

**Comment by lzap** [comment]
Hmm it failed.

**Comment by lzap** [comment]
Gimme a min to rebuild.

**Comment by lzap** [comment]
http://koji.katello.org/koji/taskinfo?taskID=220999 foreman-proxy
http://koji.katello.org/koji/taskinfo?taskID=221002 foreman

@GregSutcliffe can you provide me a guidiance of similar change for debian please? I have zero idea how to do that.

File: foreman/foreman.spec [hunk 6]
```diff
@@ -196,6 +199,17 @@ Meta Package to install hammer rubygems and its dependencies
 
 %files cli
 
+%package debug
```
Comment (domcleal) [inline foreman/foreman.spec]: There's a requires in 'foreman' for rsync, could you move it to the debug package please, as it's used by foreman-debug.

**Comment by lzap** [comment]
Fixed. Debian package is on my radar, although I'd appreciate guidiance from @GregSutcliffe. I've never created a *subpackage* in Debian... http://www.doudoulinux.org/web/english/contribute/developers/working-within-debian/article/splitting-a-debian-package.html

**Comment by domcleal** [comment]
Thanks @lzap!

Sure, there are a few of us who I'm sure would be happy to help or maybe even do it.  (Michael, Tomas, Greg and myself for instance.)

**Comment by lzap** [comment]
I think I can handle it. Thanks.

**Pull request merged**